docs/proposal/content-library-support.md

```diff
@@ -0,0 +1,257 @@
+    ...
+)
+
+func getContentLibraryItem(someargs) (itemID string, err error) {
+    var clibrary, citem string
+    u, err := url.Parse(itemToSearch)
+    if err != nil {
+    	log.Fatal(err)
+    }
+    
+    citem = u.Host
+    if u.Path != "" {
+    	clibrary = u.Host
+    	citem = strings.TrimPrefix(u.Path, "/") // Remove leading prefix /
+    }
+    
+    m := library.NewManager(restclient)
+    
+    if clibrary != "" {
+        // Try to get Content Library by ID
+    	_, err = m.GetLibraryByID(ctx, clibrary)
+    	if err != nil {
+    		// In this case, we should get all libraries with a specific name,
+    		libs, err := m.FindLibrary(ctx, library.Find{
+    			Name: clibrary,
+    		})
+    		if err != nil {
+                // ERROR on getting library			
+    		}
+    		if len(libs) != 1 {
+    			// ERROR as no library or too many libraries has been found
+    		}
+    		clibrary = libs[0]
+    	}
+    }
+    
+    items, err := m.FindLibraryItems(ctx, library.FindItem{
+    	Name:      citem,
+    	Type:      "ovf",
+    	LibraryID: clibrary,
+    })
+    if err != nil {
+        // ERROR getting the Library items	
+    }
+    
+    if len(items) != 1 {
+        // ERROR no items or too many items with same name. In this case, should be more specific and pass
+        // a content library name
+    }
+    
+    return items[0], nil
+}
+```
+
+##### VSphereVM clone changes
+
+We need to enhance the [clone](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/19c5deb79e97a8cd9d546ec39db3f29231bbcb91/pkg/services/govmomi/vcenter/clone.go#L75) function to recogize the scheme we defined, and choose appropriate API to clone the VM.
+
+Once defined if Content Library should be used, a new specific clone function should be used,
+to make the proper cloning, but using vCenter Manager client instead of template/VM manager.
+
+The "customization" function can and should be reused, and decoupled from the main clone process. Instead 
+after "customizing" it should be passed to the proper cloning function (library item or template)
+
+```go
+// Based on https://github.com/vmware-tanzu/vm-operator/blob/main/pkg/vmprovider/providers/vsphere/session/session_vm_create.go and
+// https://github.com/vmware/govmomi/blob/main/govc/library/deploy.go
+import (
+    ...
+   	"github.com/vmware/govmomi/vapi/vcenter"
+    ...
+)
+
+func deployContentLibraryAsVM(itemID string, additionalargs...) (*object.VirtualMachine, error) {
+	m := vcenter.NewManager(restclient)
+	ref, err := m.DeployLibraryItem(ctx, itemID string, vcenter.Deploy{
+		DeploymentSpec: vcenter.DeploymentSpec{
+			// Add Spec here. Contains network, storage, datastore, storagepolicy
+		},
+        Target: vcenter.Target {
+            // Add target spec here. Contains selected resource Pool, hostID, VMFolder
+        },
+	})
+    if err != nil {
+        // ERROR deploying the content library item
+    }
+
+	finder := find.NewFinder(vim25Client)
```

Comment:
```suggestion
	finder := find.NewFinder(vim25Client)
```

I assume `DeployLibraryItem` is not instantly ready and would result in a task on vsphere? so finding the object is after the task did run right? And the logic below here should then be the same as for the normal clone?

Reply:
So, it appears to not return a task, I'm checking here

Reply:
It does not show the task in the deploy result though it does create a vCenter task, you can find it by query vCenter task API, and may check how VM operator handle it.